src/libstrongswan/plugins/openxpki/openxpki_ocsp_responder.c

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright (C) 2023 Andreas Steffen, strongSec GmbH
+ *
+ * Copyright (C) secunet Security Networks AG
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation; either version 2 of the License, or (at your
+ * option) any later version.  See <http://www.fsf.org/copyleft/gpl.txt>.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * for more details.
+ */
+
+#include <library.h>
+#include <utils/debug.h>
+
+#include "openxpki_ocsp_responder.h"
+
+typedef struct private_openxpki_ocsp_responder_t private_openxpki_ocsp_responder_t;
+
+/**
+ * Private Data of a openxpki_ocsp_responder_t object.
+ */
+struct private_openxpki_ocsp_responder_t {
+
+	/**
+	 * Public data
+	 */
+	openxpki_ocsp_responder_t public;
+
+	/**
+	 * OpenXPKI certificate database
+	 */
+	database_t *db;
+};
+
+METHOD(ocsp_responder_t, get_status, cert_validation_t,
+	private_openxpki_ocsp_responder_t *this, certificate_t *cacert,
+	chunk_t serial_number, time_t *revocation_time, crl_reason_t *reason)
+{
+	cert_validation_t validation = VALIDATION_FAILED;
+	int rev_time;
+	const int max_decimals = 49;
+	char serial[max_decimals + 1], authKeyId[60];
+	char *status, *reason_code;
+	chunk_t subjectKeyId;
+	public_key_t *public;
+	enumerator_t *e;
+	bool success;
+
+	/* convert serialNumber from binary to decimal as required for the DB query.
+	 * check for a potential overflow since the database table field supports
+	 * up to 49 decimal digits only.
+	 */
+	if ((int)(2.4083 * (double)serial_number.len) + 1 > max_decimals)
```

Comment:
As mentioned in our discussion, this could just be a comparison against a compile-time constant:

```suggestion
	const int max_decimals = 49;
	const int max_bytes = max_decimals / 2.4083;
	char serial[max_decimals + 1], authKeyId[60];
	char *status, *reason_code;
	chunk_t subjectKeyId;
	public_key_t *public;
	enumerator_t *e;
	bool success;

	/* convert serialNumber from binary to decimal as required for the DB query.
	 * check for a potential overflow since the database table field supports
	 * up to 49 decimal digits only.
	 */
	if (serial_number.len > max_bytes)
```

Reply:
Implemented as  a constant